### ISSUE TITLE
BUG: Error when stretch is specified before image is added

### ISSUE THREAD

**Issue by valpasq** [bullocke/gee-estimation-tools#6]
Selecting a new RGB combination before a point in the time series is clicked produces the following error:
`: Layer error: Image.visualize: No band named 'NIR'. Available band names: [constant].`

Error handling should be happening around L191, but is not working correctly:
```
    chart.onClick(function(coords) {
      if (coords) {
        IMAGEADDED = true
        getImageRegion(ee.Feature(region).geometry(), coords)
      }
    });
````

**Comment by valpasq** [comment]
I think I fixed this by updating the `select_stretch` function:

```
var select_stretch = ui.Select({
    items: Object.keys(stretches),
    onChange: function(key) {
      if (IMAGEADDED === true) {
        visParams = stretches[key];
        updateImage(visParams)
      }
      else {
        visParams = stretches[key]
      }
    }
  });
```

Apparently, this was an issue of logic statements... Originally, the `if` was based on `IMAGEADDED = true`, which evaluates to `true` if the variable exists (regardless of value). The equal value logical would be `IMAGEADDED === true`. But if we only update the visualization parameters once an image has been added, they won't update if the user changes the RGB combo before they click a TS point (resulting in the default RGB instead of the user-selected value from the button). So I added an `else` statement to update the `visParams` values any time the button is changed. No more missing band error, and everything I've tried seems to be working as intended.

Bonus: The suspect code block in the original issue sets `IMAGEADDED` to `true` when a plot point is clicked, then gets the associated image with the `getImageRegion` function. Since this `chart.onClick` is replicated in a couple of places, I moved the `IMAGEADDED = true` to `getImageRegion` such that if you are getting an image, you are setting `IMAGEADDED = true`. Seemed to make sense and cut out a couple of duplicate lines.

Would probably be good to give the fix a test and make sure nothing else broke. Can close once we've confirmed all is well.
